infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -275,12 +271,15 @@ public static void main(String[] args) {
         // Create a common Server Context for all servers to access.
         serverContext = new ServerContext(opts, router);
 
+        // Generate a node ID if necessary.
+        generateNodeId(serverContext);
```

Comment:
Should this be generated here?
Or can we move it down in the ServerContext creation.
This way we will have a similar behaviour between our tests and real servers.

Reply:
Ok - moved it to server context. Going to wait for #1061 to make meaningful tests.